### PR TITLE
Replaced the twilio adapter with Generic SMS adaptor

### PR DESCRIPTION
**Pull request by pankaj-ag**
Thank you for the library. 

We want to use it in one of our projects but we are using Gupshup instead of Twilio. So that's why I did some modification in the code and make it a bit more generic. 

**Changes:**
1. Updated the readme file with changes
2. Renamed Twilio adapter with SMS adaptor
3. Moved Twilio adaptor from package config to the application config
4. Updated ex_twilio version to 0.8.1
5. Updated test cases for the same

Please have a look at the code and if that makes sense to you merge it. It will help us to move on this repo instead of forked one.

### CODE REVIEW

**Comment by dlobo** [comment]
This will also enable us to publish our code on hexdocs.pm if this package is also updated in hexdocs.pm. Thank you for your work

**Comment by dlobo** [comment]
any thoughts or update on this? we'd like to release on Hexdocs.pm and this is a blocker. A response would be super appreciated 👍

**Comment by samueljmurray** [comment]
Hi @dlobo and @pankaj-ag - sorry for the silence. I don't work at madebymany any more and I don't think this is on their radar at the moment. I've just contacted someone there about getting permissions or ownership on the repo so I can help you out. I'll be in touch soon

**Comment by dlobo** [comment]
great thanx for the response @samueljmurray

**Comment by samueljmurray** [comment]
@dlobo @pankaj-ag I have access to this repo now.

It looks like there's more to do to disentangle this library from Twilio so that it would integrate easily with other SMS adapters. For example, [here](https://github.com/madebymany/passwordless_auth/pull/3/files#diff-44771d5e8b57fb51a8c65940464e5edcR86) we're assuming that the SMS adapter has a module called `Message` that has function called `create`, and that it takes request options in the same way as the ExTwilio library.

Could you give me an example of how you've integrated this with Gupshup? Do you have a module you wrote that makes calls to the Gupshup API?

**Comment by pankaj-ag** [comment]
@samueljmurray thank you for your effort and response. 

Yes, you are right that on the assumption of the Message module. The reason behind this is that this is the default structure for the Twilio ex package, and we don't want to change a lot of code in this package. So we kept it like that. (I have updated the readme as well).  So this update won't impact the existing behavior.


Yes on our end we defined the SMS adapter in our config and wrote a module that makes the call on the gupshup API. 

You can have a look at the code here. 

https://github.com/glific/glific/blob/master/config/config.exs#L50
https://github.com/glific/glific/blob/master/lib/glific/providers/gupshup/message.ex#L163

Please let me know if you have any other queries or suggestions on this.

**Comment by dlobo** [comment]
any progress on this @samueljmurray

**Comment by samueljmurray** [comment]
Hi had a think about this and realised that we should just take the responsibility for sending SMS out of this library. I think it simplifies things a lot, both for the way the library works and for how people use it.

I've created a branch: https://github.com/madebymany/passwordless_auth/tree/v0.3.0

Could you have a read through the new README there, particularly the [Usage](https://github.com/madebymany/passwordless_auth/tree/v0.3.0#usage) section and try it out with your codebase?

```elixir
{:passwordless_auth, github: "madebymany/passwordless_auth#v0.3.0"}
```

**Comment by dlobo** [comment]
awesome. we are taking the rest of the week off, so will get back to you on monday with feedback on this

**Comment by pankaj-ag** [comment]
@samueljmurray thanks for the updates. I tried with the #v0.3.0 and it's working great. 
Can you please merge the branch to master now?

File: README.md [hunk 1]
```diff
@@ -26,15 +26,19 @@ end
...
 ```elixir
+config :passwordless_auth, sms_adapter: ExTwilio
+
 config :ex_twilio,
   account_sid: "TWILIO_ACCOUNT_SID",
   auth_token: "TWILIO_AUTH_TOKEN",
   workspace_sid: "TWILIO_WORKSPACE_SID" # optional
 ```
 
+Note: We expact that the custome SMS adapter has a module called Message with create function.
```
Comment (raffij) [inline README.md:40]:
```suggestion
Note: We expect the custom SMS adapter to have a module called Message with a create function.
```

File: README.md [hunk 2]
```diff
@@ -26,15 +26,19 @@ end
 
 ## Configuration
 
-PasswordlessAuth depends on [ExTwilio config](https://github.com/danielberkompas/ex_twilio) so you need to set ExTwilio config in your `config/config.exs` file:
+PasswordlessAuth uses [ExTwilio](https://github.com/danielberkompas/ex_twilio) as a default sms adapter. So if you want to use that then you need to set ExTwilio config in your `config/config.exs` file:
```
Comment (raffij) [inline README.md:29]:
```suggestion
PasswordlessAuth uses [ExTwilio](https://github.com/danielberkompas/ex_twilio) as a default sms adapter. If you wish to use ExTwilio you'll also need to set the following configuration in your `config/config.exs` file:
```

**Review by raffij — COMMENTED** [review]
Made some small changes to the readme text used.

**Comment by raffij** [comment]
Seen sam's version, so disregard the comments I made in the review.

**Comment by dlobo** [comment]
bringing this back up. any chance of a merge and hex release this week

**Comment by samueljmurray** [comment]
That's up on hex now: https://hex.pm/packages/passwordless_auth/0.3.0 Thanks for contributing 🙏

**Comment by dlobo** [comment]
great. awesome. thanx for creating the package. Our code that uses the package is also up on hex now 👍 

[Glific](https://hexdocs.pm/glific/readme.html)

**Pull request closed without merging**